lib/libproc/proc_regs.c

```diff
@@ -35,14 +35,32 @@ __FBSDID("$FreeBSD$");
 #include <sys/types.h>
 #define	_WANT_MIPS_REGNUM
 #include <sys/ptrace.h>
-
+#ifdef __mips__
+#include <machine/regnum.h>
+#endif
 #include <err.h>
 #include <stdio.h>
 #include <string.h>
 #include <errno.h>
 
 #include "_libproc.h"
 
+#if __has_feature(capabilities)
+#include <cheri/cheric.h>
+#include <cheri/cheri.h>
+
+static int
+get_pcc_base(struct proc_handle *phdl)
+{
+	struct capreg capregs;
+
+	if (ptrace(PT_GETCAPREGS, proc_getpid(phdl), (caddr_t)&capregs, 0) < 0)
+		return (-1);
```

Comment:
Returning `-1` without checking it at regvalue assignment may result in unexpected decrement/increment of `regvalue`.
Also `int` may be not wide enough for a vaddr, better use `vaddr_t`, perhaps you want to pass the `regvalue` argument and fill it directly with `(__cheri_addr vadd_t)pcc_value`?